File: update_blades.py
import os
import re

for root, _, files in os.walk('resources/views'):
    for file in files:
        if file.endswith('.php'):
            filepath = os.path.join(root, file)
            with open(filepath, 'r', encoding='utf-8') as f:
                content = f.read()
            new_content = re.sub(r'asset\(''images/'' \. (\$[a-zA-Z0-9_\->\[\]'']+) ?\)', r'str_starts_with(\1, ''http'') ? \1 : asset(''images/'' . \1)', content)
            if new_content != content:
                with open(filepath, 'w', encoding='utf-8') as f:
                    f.write(new_content)
                print(f"Updated {filepath}")

File: update_blades.php
<?php
$views = new RecursiveIteratorIterator(new RecursiveDirectoryIterator('resources/views'));
foreach ($views as $view) {
    if ($view->isFile() && $view->getExtension() === 'php') {
        $content = file_get_contents($view->getPathname());
        $newContent = preg_replace(''~asset\(''images/'' \. (\$[a-zA-Z0-9_\->\[\]'']+) ?\)~'', ''\Illuminate\Support\Str::startsWith($1, ["http://", "https://"]) ? $1 : asset("images/" . $1)'', $content);
        if ($newContent !== $content) {
            file_put_contents($view->getPathname(), $newContent);
            echo "Updated " . $view->getPathname() . "\n";
        }
    }
}
